### ISSUE TITLE
Adventure pals proton crashes 

### ISSUE THREAD

**Issue by monterrr** [flathub/com.valvesoftware.Steam#502]
**Game information**
Adventure pals.

**Distribution name and version where applicable**
Fedora 31

**Flatpak info**
Flatpak 1.4.3
nvidia-435-21

**Problem description**
Adventure pals proton game crashes, if using SHARED_LIBRARY_GUARD=0 it no longer crashes.

**Does this issue reproduce with native Steam**
Have not tested

**Comment by nanonyme** [comment]
Do you have any idea which Proton version this game is using?

**Comment by monterrr** [comment]
Tested with 4.11 and 4.2, crashes on very specific game places with in game video.

**Comment by nanonyme** [comment]
@gasinvein I'm leaning towards removing the general Proton blocklist. https://github.com/flathub/com.valvesoftware.Steam/blob/beta/resources/blocklist/proton.yml
There's just too many issues with these blanket things. Do most games even really need those?

**Comment by gasinvein** [comment]
@nanonyme I agree, it looks like that using shared-library-guard introduces more problems than it does solve. Yet still I don't understand why this happens.

**Comment by nanonyme** [comment]
Shared-library-guard has worked just fine for games where individual rules are added. Problems have always been related with these blanket rules like with Proton. Its original purpose was to fix broken games, I think the Proton rule is not really about that for everything it covers.

**Comment by gasinvein** [comment]
@nanonyme It seems like rules doesn't really matter, some Proton games crash with shared-library-guard enabled regardless of the rules. I.e. it's not related to FAudio.

**Comment by nanonyme** [comment]
That is certainly true. What I meant is not all games affected by this rule necessarily need the extra codecs.

**Comment by gasinvein** [comment]
Yeah. Maybe we could apply s-l-g rules to Windows binaries? They run just like all other Linux processes after all.
